CoreFunctionality/src/au/gov/asd/tac/constellation/functionality/startup/ConstellationLAFSettings.java

```diff
@@ -0,0 +1,502 @@
+            if ( (keyFilter == null || uiKey.toString().contains(keyFilter)) &&
+                 (valueFilter == null || UIManager.getLookAndFeelDefaults().get(uiKey).toString().contains(valueFilter)) ) {
+                msg = ">> :::: " + uiKey + " = " + UIManager.getLookAndFeelDefaults().get(uiKey);
+                LOGGER.info(msg);                
+            }
+        }
+    }    
+    
+    /**
+     * Custom painter required for Nimbus Look and Feel to override the default
+     * colors used for tabs.
+     * Will fill the background color of the tab with a gradient transition of
+     * color between the color(s) specified in the constructor call.
+     */
+    public static class NimbusCustomGradientTabPainter<T> implements Painter<T> {
+
+        private final Color upperColor;
+        private final Color lowerColor;
+
+        /**
+         * Default constructor will use a White to Black color gradient.
+         */
+        public NimbusCustomGradientTabPainter() {
+            upperColor = Color.WHITE;
+            lowerColor = Color.BLACK;
+        }
+
+        /**
+         * Single color constructor will use the specified color as the upper
+         * (lighter shade) color and calculate a darker shade of the same color
+         * to use as the bottom shade.
+         * The background color of the tab will be painted with a gradient
+         * transition between the light and dark shades of the color.
+         * @param baseColor upper/lighter shade of color to be used in the color gradient
+         */
+        public NimbusCustomGradientTabPainter(final Color baseColor) {
+            if (baseColor != null) {
+                upperColor = baseColor;
+                float[] baseHSB = new float[3];
+                Color.RGBtoHSB(baseColor.getRed(), baseColor.getGreen(), baseColor.getBlue(), baseHSB);
+                lowerColor = new Color(Color.HSBtoRGB(baseHSB[0], baseHSB[1], baseHSB[2] * 2 / 3));
+            } else {
+                upperColor = Color.WHITE;
+                lowerColor = Color.BLACK;
+            }
+        }
+
+        /**
+         * Dual color constructor allows top and bottom shades to be specified
+         * individually. The Tab background will be painted as a gradient from
+         * top to bottom, transitioning between the specified colors.
+         * @param topShadeColor Color to paint at top of Tab
+         * @param bottomShadeColor Color to paint at bottom of Tab
+         */
+        public NimbusCustomGradientTabPainter(final Color topShadeColor, final Color bottomShadeColor) {
+            if (topShadeColor != null) {
+                upperColor = topShadeColor;
+            } else {
+                upperColor = Color.WHITE;
+            }
+            if (bottomShadeColor != null) {
+                lowerColor = bottomShadeColor;
+            } else {
+                lowerColor = Color.BLACK;
+            }
```

Comment:
are you able to use ternary statements for these 2 `if-elses` instead?

Reply:
Done